src/apps/reservation-list/modal/reservation-details/reservation-details-buttons.tsx

```diff
@@ -1,71 +1,33 @@
-import React, { FC, useCallback } from "react";
-import { useQueryClient } from "react-query";
+import React, { FC } from "react";
 import { useText } from "../../../../core/utils/text";
 import { MaterialProps } from "../../../loan-list/materials/utils/material-fetch-hoc";
-import {
-  useDeleteReservations,
-  getGetReservationsV2QueryKey
-} from "../../../../core/fbs/fbs";
-import DeleteReservationModal from "../delete-reservation/delete-reservation-modal";
-import { getModalIds } from "../../../../core/utils/helpers/general";
-import { useModalButtonHandler } from "../../../../core/utils/modal";
 
 export interface ReservationDetailsButtonProps {
   reservationId: number;
   numberInQueue?: number | null;
+  openReservationDeleteModal: (deleteId: number | string | null) => void;
 }
 
 const ReservationDetailsButton: FC<
   ReservationDetailsButtonProps & MaterialProps
-> = ({ reservationId, numberInQueue }) => {
+> = ({ reservationId, numberInQueue, openReservationDeleteModal }) => {
   const t = useText();
-  const queryClient = useQueryClient();
-  const modalIds = getModalIds();
-  const { mutate } = useDeleteReservations();
-  const { open, close } = useModalButtonHandler();
-  const modalId = `${modalIds.deleteReservation}${reservationId}`;
-
-  const deleteReservation = useCallback(() => {
-    if (reservationId) {
-      mutate(
-        {
-          params: { reservationid: [reservationId] }
-        },
-        {
-          onSuccess: () => {
-            queryClient.invalidateQueries(getGetReservationsV2QueryKey());
-            close(modalId);
-          },
-          // todo error handling, missing in figma
-          onError: () => {
-            close(modalId);
-          }
-        }
-      );
-    }
-  }, [close, modalId, mutate, queryClient, reservationId]);
 
   return (
-    <>
-      <div className="modal-details__buttons">
-        {numberInQueue && numberInQueue > 0 && (
-          <div className="my-8 mx-16 text-body-medium-regular">
-            {t("reservationDetailsOthersInQueueText")}
-          </div>
-        )}
-        <button
-          type="button"
-          onClick={() => open(modalId)}
-          className="btn-primary btn-filled btn-small arrow__hover--right-small"
-        >
-          {t("reservationDetailsButtonRemoveText")}
-        </button>
-      </div>
-      <DeleteReservationModal
-        deleteReservation={deleteReservation}
-        id={modalId}
-      />
-    </>
+    <div className="modal-details__buttons">
+      {numberInQueue && numberInQueue > 0 && (
+        <div className="my-8 mx-16 text-body-medium-regular">
+          {t("reservationDetailsOthersInQueueText")}
+        </div>
+      )}
+      <button
+        type="button"
+        onClick={() => openReservationDeleteModal(reservationId)}
+        className="btn-primary btn-filled btn-small arrow__hover--right-small"
+      >
+        {t("reservationDetailsButtonRemoveText")}
+      </button>
```

Comment:
Cant you use one of the button components?
https://github.com/danskernesdigitalebibliotek/dpl-react/tree/develop/src/components/Buttons

Reply:
Oki doki